posixlib/src/main/scala/scala/scalanative/runtime/SocketHelpers.scala

```diff
@@ -107,33 +104,32 @@ object SocketHelpers {
 
   def hostToIp(host: String): Option[String] = {
     Zone { implicit z =>
-      var hints = alloc[addrinfo]
-      var ret   = alloc[Ptr[addrinfo]]
-
-      var ipstr = alloc[CChar]((INET6_ADDRSTRLEN + 1).toUInt)
-      libc.memset(hints.rawptr, 0, sizeof[addrinfo])
+      val hints = alloc[addrinfo]
       hints.ai_family = AF_UNSPEC
-      hints.ai_socktype = 0
-      hints.ai_next = null
+
+      val ret = alloc[Ptr[addrinfo]]
+
+      val ipstr = alloc[CChar]((INET6_ADDRSTRLEN + 1).toUInt)
 
       val status = getaddrinfo(toCString(host), null, hints, ret)
       if (status != 0)
         return None
 
-      var addr = alloc[Byte]
-      if ((!ret).ai_family == AF_INET) {
-        addr = (!ret).ai_addr
-          .asInstanceOf[Ptr[sockaddr_in]]
-          .sin_addr
-          .toPtr
-          .asInstanceOf[Ptr[Byte]]
-      } else {
-        addr = (!ret).ai_addr
-          .asInstanceOf[Ptr[sockaddr_in6]]
-          .sin6_addr
-          .toPtr
-          .asInstanceOf[Ptr[Byte]]
-      }
+      val addr =
+        if ((!ret).ai_family == AF_INET) {
+          (!ret).ai_addr
+            .asInstanceOf[Ptr[sockaddr_in]]
+            .sin_addr
+            .toPtr
```

Comment:
Do you have to call `toPtr` here?

Reply:
That code, and its sibling in the else branch is prior art. Much as my fingers itched, I tried not
to disturb it.  

git blame says that it was written by densh.  sin_addr is a structure `struct in_addr  sin_addr` 
(Open Group netinet/in.h).  Now converting a structure to a Ptr is somewhat strange.
To my thinking, converting the enclosed field s_addr which is some form of an Int makes
more sense to a Ptr makes more sense.

Given a dispensation from "only memset changes", I can make the s_addr change on both 
branches and post it for review.   Please advise.

So much prior art, so little time....

Reply:
I tried it out and it works but `CStruct1` has a `toPtr` that returns a typed pointer so that is hard to beat.